exchange.c

```diff
@@ -254,46 +249,39 @@ static int domain_exchange_once(int (*layoutfunc)(int p), int* toGo, int * toGoS
     if(N_bh_slots > All.MaxPartBh)
         endrun(787878, "Task=%d N_bh=%d All.MaxPartBh=%d\n", ThisTask, N_bh_slots, All.MaxPartBh);
 
-    MPI_Alltoallv_sparse(partBuf, count, offset, MPI_TYPE_PARTICLE,
-                 P, count_recv, offset_recv, MPI_TYPE_PARTICLE,
+    MPI_Alltoallv_sparse(partBuf, count[0], offset[0], MPI_TYPE_PARTICLE,
+                 P, count_recv[0], offset_recv[0], MPI_TYPE_PARTICLE,
                  MPI_COMM_WORLD);
     walltime_measure("/Domain/exchange/alltoall");
 
-    MPI_Alltoallv_sparse(sphBuf, count_sph, offset_sph, MPI_TYPE_SPHPARTICLE,
-                 SphP, count_recv_sph, offset_recv_sph, MPI_TYPE_SPHPARTICLE,
+    MPI_Alltoallv_sparse(sphBuf, count[1], offset[1], MPI_TYPE_SPHPARTICLE,
+                 SphP, count_recv[1], offset_recv[1], MPI_TYPE_SPHPARTICLE,
                  MPI_COMM_WORLD);
     walltime_measure("/Domain/exchange/alltoall");
 
-    MPI_Alltoallv_sparse(bhBuf, count_bh, offset_bh, MPI_TYPE_BHPARTICLE,
-                BhP, count_recv_bh, offset_recv_bh, MPI_TYPE_BHPARTICLE,
+    MPI_Alltoallv_sparse(bhBuf, count[2], offset[2], MPI_TYPE_BHPARTICLE,
```

Comment:
With a bit more work we can move these duplicated into a loop as well. I can take care of it after merging this.

Reply:
Thanks. I did not see a way to do that!